src/components/common/formFields/Widgets.tsx

```diff
@@ -0,0 +1,366 @@
+    return (
+        <div className={`styled-form-box ${showBox ? 'show-box' : ''}`}>
+            <h2 className="fs-14 fw-6 p-0 m-0 lh-20">{title}</h2>
+            {description && <p className="fs-11 fw-4 p-0 mt-4 lh-16">{description}</p>}
+            <hr />
+            {children}
+        </div>
+    )
+}
+
+interface StyledInputPropsType extends WidgetPropsType {
+    value: string
+    pattern?: string
+}
+
+export const StyledInput = ({
+    title,
+    value,
+    description,
+    rootClassName,
+    onChange,
+    onBlur,
+    minLength,
+    maxLength,
+    pattern,
+    isRequired,
+    placeholder,
+    errorMessage,
+}: StyledInputPropsType): JSX.Element => {
+    return (
+        <StyledField
+            title={title}
+            description={description}
+            rootClassName={rootClassName}
+            isRequired={isRequired}
+            errorMessage={errorMessage}
+        >
+            <input
+                placeholder={placeholder && placeholder}
+                name={title.replace(/\s/g, '').toLowerCase()}
+                className="form__input"
+                value={value as string}
+                onChange={onChange}
+                onBlur={onBlur}
+                autoComplete="off"
+                required={isRequired}
+                minLength={minLength}
+                maxLength={maxLength}
+                pattern={pattern}
+            />
+        </StyledField>
+    )
+}
+
+export const StyledTextarea = ({
+    title,
+    value,
+    description,
+    rootClassName,
+    onChange,
+    onBlur,
+    minLength,
+    maxLength,
+    isRequired,
+    placeholder,
+    errorMessage,
+}: StyledInputPropsType): JSX.Element => {
+    return (
+        <StyledField
+            title={title}
+            description={description}
+            rootClassName={rootClassName}
+            isRequired={isRequired}
+            errorMessage={errorMessage}
+        >
+            <textarea
+                className="form__input form__textarea"
+                name={title.replace(/\s/g, '').toLowerCase()}
+                value={value as string}
+                placeholder={placeholder && placeholder}
+                onChange={onChange}
+                onBlur={onBlur}
+                autoComplete="off"
+                required={isRequired}
+                minLength={minLength}
+                maxLength={maxLength}
+            />
+        </StyledField>
+    )
+}
+
+interface SliderPropsType extends WidgetPropsType {
+    unit: string
+    defaultValue?: number
+    onInput?: (e) => void
+    hideValueInput?: boolean
+}
+
+const sliderRangeUpdate = (e, value: number, min: number, max: number) => {
+    const newValue = ((value - min) / (max - min)) * 100
+    e.style.background = `linear-gradient(to right, var(--B500) 0%, var(--B500) ${newValue}%, var(--N200) ${newValue}%, var(--N200) 100%)`
+}
+
+export const Slider = ({
```

Comment:
this component is going way more lengthy. Can we break it down in sub components?

Reply:
As discussed, this a common component containing different widgets as small comps.